Test/Unit/Service/ReplicaSettingsHandlerTest.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Test\Unit\Service;
+
+use Algolia\AlgoliaSearch\Api\Data\IndexOptionsInterface;
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Algolia\AlgoliaSearch\Service\AlgoliaConnector;
+use Algolia\AlgoliaSearch\Service\ReplicaSettingsHandler;
+use PHPUnit\Framework\TestCase;
+
+class ReplicaSettingsHandlerTest extends TestCase
+{
+    protected ?AlgoliaConnector $connector = null;
+
+    protected ?ConfigHelper $config = null;
+
+    protected ?IndexOptionsInterface $indexOptions = null;
+
+    private ?ReplicaSettingsHandler $handler = null;
+
+    protected function setUp(): void
+    {
+        $this->connector = $this->createMock(AlgoliaConnector::class);
+        $this->config = $this->createMock(ConfigHelper::class);
+        $this->indexOptions = $this->createMock(IndexOptionsInterface::class);
+
+        $this->handler = new ReplicaSettingsHandlerTestable($this->connector, $this->config);
+    }
+
+    public function testSetSettingsWithForwardingEnabledAndMixedSettings(): void
+    {
+        $storeId = 1;
+        $settings = [
+            'customRanking' => ['desc(price)'],
+            'attributesToRetrieve' => ['name', 'price']
+        ];
+
+        $this->indexOptions->method('getStoreId')->willReturn($storeId);
+        $this->config->method('shouldForwardPrimaryIndexSettingsToReplicas')
+            ->with($storeId)
+            ->willReturn(true);
+
+        $invocations = $this->exactly(2);
+        $this->connector->expects($invocations)
+            ->method('setSettings')
+            ->willReturnCallback(
+                function($indexOptions, $indexSettings, $forwardToReplicas, $mergeSettings, $mergeFrom) use ($invocations) {
+                    switch ($invocations->numberOfInvocations()) {
```

Comment:
This doesn't seem to work with PHPUnit 9, don't we want to keep compatibility with it until 2.4.7 support ends ?

Reply:
All green now :)